spec/api.spec.ts

```diff
@@ -46,7 +46,7 @@ describe("getParentOfType", () => {
   });
 
   test("returns the proper root for class model instance", () => {
-    const m = new TestClassModel(TestModelSnapshot);
+    const m = TestClassModel.createReadOnly(TestModelSnapshot);
```

Comment:
This was an accidental holdover from the original API I was thinking about using for class models. Since then we switched to making them conform to the normal setup that all the other type uses, which is static `.create` / `.createReadOnly`, and the constructor is effectively private. Some of these optimizations required changing the constructor signature, which I think is fine as this was just an accidental slip through.